### PR TITLE
Add post_process_semantic_segmentation method to SegFormer

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
Adds a post_process_semantic_segmentation method to `SegFormerFeatureExtractor` with optional resizing. This model doesn't support instance or panoptic segmentation. 

I will open separate PRs to make sure the naming and outputs of post_process methods of segmentation models are consistent.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/segformer/feature_extraction_segformer.py [hunk 1]
```diff
@@ -211,3 +214,46 @@ def __call__(
...
+            outputs ([`SegformerForSemanticSegmentation`]):
+                Raw outputs of the model.
+            target_sizes (`torch.Tensor` of shape `(batch_size, 2)` or `List[Tuple]` of length `batch_size`, *optional*):
+                Torch Tensor (or list) corresponding to the requested final size (h, w) of each prediction. If left to
+                None, predictions will not be resized.
+
+        Returns:
+            `List[torch.Tensor]`: A list of `torch.Tensor` of length `batch_size`, where each item is a semantic
+            segmentation map of of the corresponding target_sizes entry (if `target_sizes` is specified). Each entry of
+            each `torch.Tensor` correspond to a semantic class id.
```
Comment (NielsRogge) [inline src/transformers/models/segformer/feature_extraction_segformer.py:233]: It seems that if no `target_sizes` are specified, a single `torch.Tensor` is returned, whereas if target_sizes is specified, a list is returned.

Could we update this docstring, or make the output consistent?

Comment (alaradirik) [reply src/transformers/models/segformer/feature_extraction_segformer.py:233]: Thanks, I made the output consistent!

**Comment by alaradirik** [comment]
FYI, I will also need to open a PR to edit `ImageSegmentationPipeline` after making sure all post-processing methods across segmentation models are consistent in terms of naming and functionality.

**Review by NielsRogge — APPROVED** [review]
LGTM

**Comment by LysandreJik** [comment]
Hey @alaradirik, please also ping a core maintainer for review before merging PRs.

**Pull request merged**